### PR TITLE
Extract kubelet node status into separate file

### PR DESCRIPTION
**Pull request by pmorie**
Extract kubelet node status management into a separate file as a continuation of the kubelet code simplification effort.

### CODE REVIEW

**Comment by pmorie** [comment]
@vishh PTAL

**Comment by pmorie** [comment]
cc @kubernetes/sig-node

File: pkg/kubelet/util.go [hunk 1]
```diff
@@ -118,3 +119,12 @@ func allowHostIPC(pod *api.Pod) (bool, error) {
 	}
 	return false, nil
 }
+
+// dirExists returns true if the path exists and represents a directory.
+func dirExists(path string) bool {
```
Comment (yujuhong) [inline pkg/kubelet/util.go]: suggestion: this function seems general enough. How about moving it to `pkg/kubelet/util/` ?

Comment (pmorie) [inline pkg/kubelet/util.go]: I'm down w/ that, okay to do in follow-up?

Comment (yujuhong) [inline pkg/kubelet/util.go]: yes.

**Pull request merged**